java-checks/src/main/java/org/sonar/java/checks/SunPackagesUsedCheck.java

```diff
@@ -79,13 +79,15 @@ private static boolean isActuallySunPackage(MemberSelectExpressionTree tree) {
     // Check if the expression's type actually comes from a sun.* package
     // This prevents false positives when a variable is named "sun"
     var type = tree.expression().symbolType();
-    if (type.isUnknown()) {
-      // If we can't determine the type, we should still check to avoid missing real issues
-      // In this case, rely on the string-based check
-      return true;
+    if (!type.isUnknown()) {
+      // When we have type information, use it to determine if it's actually a sun.* package
+      String fullyQualifiedName = type.fullyQualifiedName();
+      return fullyQualifiedName.startsWith("sun.");
     }
-    String fullyQualifiedName = type.fullyQualifiedName();
-    return fullyQualifiedName.startsWith("sun.");
+    // When type is unknown (non-compiling code), we can't use semantic analysis
+    // In this case, rely on the string-based check which has already been done
+    // This means we may miss some FPs in non-compiling code, but we'll catch them with proper semantic analysis
+    return true;
```

Comment:
This looks like the same logic as before, with the condition reversed. In my opinion, if we don't have semantic information available, we should return `false` to prevent annoying FPs in our users' code, e.g. a user on SQC, with AutoScan, is going to get FPs on variables called "sun"